_sips/minutes/2020-03-11-minutes.md

```diff
@@ -0,0 +1,220 @@
+6. Review dependent and polymorphic function types
+
+## Date and Location
+
+The meeting took place on the 11th March 2020 throughout the day at EPFL in Lausanne, Switzerland, and Zoom.
+
+The meeting wasn't broadcast.
+
+Minutes were taken by Dale Wijnand.
+
+## Committee Attendees
+
+* Martin Odersky ([@odersky](https://github.com/odersky)), EPFL
+* Adriaan Moors ([@adriaanm](https://github.com/adriaanm)), Lightbend
+* Guillaume Martres ([@smarter](https://github.com/smarter)), EPFL
+* Seth Tisue ([@SethTisue](https://github.com/SethTisue)), Lightbend
+* Sébastien Doeraene ([@sjrd](https://github.com/sjrd)), Scala Center
+
+## Sitting in
+
+* Jamie Thompson ([@bishabosha](https://github.com/bishabosha)), Scala Center
+* Lukas Rytz ([@lrytz](https://twitter.com/lrytz)), Lightbend
+* Nicolas Stucki ([@nicolasstucki](https://github.com/nicolasstucki)), EPFL
+* Darja Jovanovic ([@darjutak](https://github.com/darjutak)), Process Lead
+* Dale Wijnand ([@dwijnand](https://twitter.com/dwijnand)), secretary
+
+## Not present
+
+* Heather Miller ([@heathermiller](https://github.com/heathermiller)), CMU
+* Iulian Dragos ([@dragos](https://github.com/dragos)), Triplequote
+* Josh Suereth ([@jsuereth](https://github.com/jsuereth)), independent
+* Miles Sabin ([@milessabin](https://github.com/milessabin)), independent
+
+Miles notified the committee that he wouldn't be available.  Josh, Heather, and Iulian were all unable to
+participate because of coronavirus-related travel restrictions and disruptions.
+
+## Proceedings
+
+### Expectations
+
+* Martin: The plan is still to release Scala 3 this year, at the end of December.
+* Martin: However, TASTy won't be stable until 3.1 - it won't be a 3.0 promise.
+* Martin: Ideally the features would be voted on with the following 3 outcomes
+
+* accepted
+* accepted, but put behind some kind of "preview" flag, like Java is doing
+* rejected
+* postponed to after 3.0
+
+### Review a "easy" dotty features/changes
+
+This is the list, with any notable comments added.
+
+* Context functions
+* Drop auto application
+* Drop auto tupling
+    * Martin: this is probably the most breaking change, so weakly against it
+    * Seth: could it be behind `-Scala2`?  Martin: no, that would be difficult
+    * There's some difficulties in implementing a rewrite rule for it
+* Drop class shadowing
+    * Seb: this is binary breaking for existing libraries
+    * Seth: let's make it strong (not under -Xlint) w/ opt-out
+* Drop compound types
+* Drop DelayedInit
+    * most of the committee is for it
+    * but a number of usages of it have been reported
+    * Guillaume: the implementation isn't a big deal, but it affects TASTy
+    * ScalaTest, specs2, and cats-effect use it
+    * Let's drop it and hopefully someone can write a compiler plugin for it
+    * Lukas: DelayedInit was the source of lots of bugs in nsc, most fixed by now, though
+* Drop do..while
+    * added because of indentation syntax
```

Comment:
someone said that at first, but then Martin said it was mainly because of the new control syntax